server/src/internalClusterTest/java/org/opensearch/nodestats/NodeStatsIT.java

```diff
@@ -0,0 +1,274 @@
+        MatcherAssert.assertThat(response.getItems().length, equalTo(sizeOfIndexRequests));
+
+        for (BulkItemResponse itemResponse : response.getItems()) {
+            updateExpectedDocStatusCounter(itemResponse.getResponse());
+        }
+
+        refresh(INDEX);
+        bulkRequest.requests().clear();
+
+        for (int i = 0; i < sizeOfDeleteRequests; ++i) {
+            bulkRequest.add(new DeleteRequest(INDEX, String.valueOf(i)));
+        }
+        for (int i = 0; i < sizeOfNotFoundRequests; ++i) {
+            bulkRequest.add(new DeleteRequest(INDEX, String.valueOf(25 + i)));
+        }
+
+        response = client().bulk(bulkRequest).actionGet();
+
+        MatcherAssert.assertThat(response.hasFailures(), equalTo(false));
+        MatcherAssert.assertThat(response.getItems().length, equalTo(sizeOfDeleteRequests + sizeOfNotFoundRequests));
+
+        for (BulkItemResponse itemResponse : response.getItems()) {
+            updateExpectedDocStatusCounter(itemResponse.getResponse());
+        }
+
+        refresh(INDEX);
+        assertDocStatusStats();
+    }
+
+    public void testNodeIndicesStatsDocStatusStatsIndex() {
+        final String INDEX = "test_index";
+        final String ID = "id";
+        {  // Testing Normal Index
+            IndexResponse response = client().index(new IndexRequest(INDEX).id(ID).source(SOURCE)).actionGet();
+            updateExpectedDocStatusCounter(response);
+
+            MatcherAssert.assertThat(response.getResult(), equalTo(DocWriteResponse.Result.CREATED));
+            assertDocStatusStats();
+        }
+        {  // Testing Missing Alias
+            updateExpectedDocStatusCounter(
+                expectThrows(
+                    IndexNotFoundException.class,
+                    () -> client().index(new IndexRequest(INDEX).id("missing_alias").setRequireAlias(true).source(SOURCE)).actionGet()
+                )
+            );
+            assertDocStatusStats();
+        }
+        {
+            // Test Missing Pipeline: Ingestion failure, not Indexing failure
+            expectThrows(
+                IllegalArgumentException.class,
+                () -> client().index(new IndexRequest(INDEX).id("missing_pipeline").setPipeline("missing").source(SOURCE)).actionGet()
+            );
+            assertDocStatusStats();
+        }
+        {  // Testing Version Conflict
+            final String docId = "version_conflict";
+
+            updateExpectedDocStatusCounter(client().index(new IndexRequest(INDEX).id(docId).source(SOURCE)).actionGet());
+            updateExpectedDocStatusCounter(
+                expectThrows(
+                    VersionConflictEngineException.class,
+                    () -> client().index(new IndexRequest(INDEX).id(docId).source(SOURCE).setIfSeqNo(1L).setIfPrimaryTerm(99L)).actionGet()
+                )
+            );
+            assertDocStatusStats();
+        }
+    }
+
+    public void testNodeIndicesStatsDocStatusStatsCDU() {
```

Comment:
Merged `create`, `delete`, `update` tests to reduced the number of clusters spawned for this integ test.

Reply:
testNodeIndicesStatsDocStatusStatsForCreateDeleteUpdate CDU is cryptic